link_linux.go

```diff
@@ -1580,9 +1586,57 @@ func LinkDeserialize(hdr *unix.NlMsghdr, m []byte) (Link, error) {
 	}
 	*link.Attrs() = base
 
+	// If the tuntap attributes are not updated by netlink due to
+	// an older driver, use sysfs
+	if link != nil && linkType == "tun" {
+		tuntap := link.(*Tuntap)
+
+		if tuntap.Mode == 0 {
+			ifname := tuntap.Attrs().Name
+			if flags, err := readSysProp(ifname, "tun_flags"); err == nil {
+
+				if (flags & unix.IFF_TUN) != 0 {
+					tuntap.Mode = unix.IFF_TUN
+				} else if (flags & unix.IFF_TAP) != 0 {
+					tuntap.Mode = unix.IFF_TAP
+				}
+
+				if flags&unix.IFF_PERSIST == 0 {
+					tuntap.NonPersist = true
+				} else {
+					tuntap.NonPersist = false
+				}
+
+			}
+			if owner, err := readSysProp(ifname, "owner"); err == nil && owner > 0 {
+				tuntap.Owner = uint32(owner)
```

Comment:
Hmmm... Are we loosing there info that the device have no owner, instead passing an info that the owner is root? The same goes for `group`.

Reply:
If the owner is root, I noticed that the sysfs interface returns -1 instead of 0 here. That's why the workaround. The sysfs interface probably needs to be modified to return 0 for root instead of -1.

Reply:
ACK.

Reply:
Then please add this explanation as a comment.